### PR TITLE
console: Handle default value associated with OptionDetails 

### LINKED ISSUE #3786: `azd init` select template panics when using default selection
1. Create empty directory
2. Run `azd init`
3. Select "Select a template"
4. Press enter. This selects the default selection of `Minimal`

Expected:
Prompt should succeed

Actual:

```
panic: runtime error: index out of range [-2]

goroutine 1 [running]:
github.com/azure/azure-dev/cli/azd/pkg/templates.PromptTemplate({0x562ddd691948, 0xc0005f25a0}, {0x562ddce922fd, 0x1a}, 0x0?, {0x562ddd69eec8, 0xc00034c750})
        /mnt/vss/_work/1/s/cli/azd/pkg/templates/template_manager.go:229 +0xc5e
github.com/azure/azure-dev/cli/azd/cmd.(*initAction).initializeTemplate(0xc00011b020, {0x562ddd691948, 0xc0005f25a0}, 0x2?)
        /mnt/vss/_work/1/s/cli/azd/cmd/init.go:299 +0x76
github.com/azure/azure-dev/cli/azd/cmd.(*initAction).Run(0xc00011b020, {0x562ddd691948, 0xc0005f25a0})
        /mnt/vss/_work/1/s/cli/azd/cmd/init.go:199 +0x6de
```

### PR DESCRIPTION
**Pull request by weikanglim**
Handle default value associated with OptionDetails so that they would match correctly.

Fixes #3786

### CODE REVIEW

**Review by ellismg — APPROVED** [review]

File: cli/azd/pkg/input/console.go [hunk 1]
```diff
@@ -708,6 +708,10 @@ func (c *AskerConsole) Select(ctx context.Context, options ConsoleOptions) (int,
 	}
 
 	surveyOptions := make([]string, len(options.Options))
+	surveyDefault := options.DefaultValue
+	surveyDefaultAsString, surveyDefaultIsString := surveyDefault.(string)
```
Comment (vhvb1989) [inline cli/azd/pkg/input/console.go:712]: do we need `surveyDefault`?

would this work?  
```suggestion
	surveyDefaultAsString, surveyDefaultIsString := options.DefaultValue.(string)
```

**Review by vhvb1989 — APPROVED** [review]
small question

File: cli/azd/pkg/input/console.go (hunk 1 above)
Comment (weikanglim) [reply cli/azd/pkg/input/console.go:712]: Was trying to preserve the semantics that we had with `surveyOptions`.. The other point of consideration was avoiding modifying the passed in `options` (but that problem still exists with the current implementation).

Comment (weikanglim) [reply cli/azd/pkg/input/console.go:712]: Just a slight readability improvement to create a local variable that captures the intent of modification, keeping as is.

**Pull request merged**